File: Preprocessing3.py
import scipy.io
import numpy as np

# Load data from .mat files
metaInputs = scipy.io.loadmat('metaInputs.mat')['metaInputs']
metaTargets = scipy.io.loadmat('metaTargets.mat')['metaTargets']

# Data with all news parameters
# 25% of data for testing
testingData = metaInputs[:, ::4]
testingClass = metaTargets[:, ::4]

# 75% of data for training
trainingData = np.copy(metaInputs)
trainingClass = np.copy(metaTargets)

for i in range(metaInputs.shape[1] - 1, -1, -1):
    if i % 4 == 0:
        trainingData = np.delete(trainingData, i, axis=1)
        trainingClass = np.delete(trainingClass, i, axis=1)

# Data with only the 4 linguistic parameters
metaInputsLing = metaInputs[[10, 11, 14, 19], :]

# 25% of data for testing
testingDataLing = metaInputsLing[:, ::4]
testingClassLing = metaTargets[:, ::4]


# 75% of data for training
trainingDataLing = np.copy(metaInputsLing)
trainingClassLing = np.copy(metaTargets)

for i in range(metaInputsLing.shape[1] - 1, -1, -1):
    if i % 4 == 0:
        trainingDataLing = np.delete(trainingDataLing, i, axis=1)
        trainingClassLing = np.delete(trainingClassLing, i, axis=1)

# Saving the variables into a .npz file
np.savez('saved_data.npz',
         testingData=testingData, testingClass=testingClass,
         trainingData=trainingData, trainingClass=trainingClass,
         testingDataLing=testingDataLing, testingClassLing=testingClassLing,
         trainingDataLing=trainingDataLing, trainingClassLing=trainingClassLing)

File: Preprocessing3.m
%% Dados
clear all
clc
load('metaInputs.mat');
load('metaTargets.mat')

%% Data with all news parameters
%25% of data for testing

testingData = metaInputs(:,1:4:end);
testingClass = metaTargets(:,1:4:end);

%75% of data for training
trainingData = metaInputs;
trainingClass = metaTargets;

for i = size(metaInputs,2):-1:1
    if ismember(i,[1:4:size(metaInputs,2)]) == 1
        trainingData(:,i) = [] ;
        trainingClass(:,i) = [];
    end
end


%% Data with only the 4 linguistic parameters
metaInputsLing = metaInputs([11 12 15 20],:);

%25% of data for testing

testingDataLing = metaInputsLing(:,1:4:end);
testingClassLing = metaTargets(:,1:4:end);

%75% of data for training
trainingDataLing = metaInputsLing;
trainingClassLing = metaTargets;

for i = size(metaInputsLing,2):-1:1
    if ismember(i,[1:4:size(metaInputsLing,2)]) == 1
        trainingDataLing(:,i)=[] ;
        trainingClassLing(:,i) = [];
    end
end

